### PR TITLE
Add pytest --integration option and tests, add install --force

### PR DESCRIPTION
**Pull request by akarve**

### CODE REVIEW

File: quilt/test/test_install.py [hunk 1]
```diff
@@ -125,3 +125,15 @@ def _mock_package(self, package, pkg_hash, contents, hashes):
     def _mock_s3(self, pkg_hash, contents):
         s3_url = 'https://example.com/%s' % pkg_hash
         self.requests_mock.add(responses.GET, s3_url, contents)
+
+# execute outside of mock scope so we get real network
```
Comment (dimaryaz) [inline quilt/test/test_install.py]: Uh, unit tests should never hit real servers. You can add "integration test", but it should be a separate file - not something that runs as a part of `pytest`.

File: quilt/test/test_install.py [hunk 2]
```diff
@@ -125,3 +125,15 @@ def _mock_package(self, package, pkg_hash, contents, hashes):
...
+def test_prod_install():
+    # public package
+    command.install('akarve/days', force=True)
+    from quilt.data.akarve import days
+    df = days.names.data()
+    assert(df.loc[3]['Day'] == 'Wednesday')
+    # TODO clean up after this test b/c tests shouldn't have side-effects
```
Comment (dimaryaz) [inline quilt/test/test_install.py]: Create a temporary directory, `chdir`, run `install`, remove the directory - same as what `QuiltTestCase` does.

File: quilt/tools/command.py [hunk 3]
```diff
@@ -410,10 +410,13 @@ def install(package, hash=None, version=None, tag=None):
...
-        overwrite = input("Overwrite? (y/n) ")
-        if overwrite.lower() != 'y':
-            return
+        if not force:
+            print("{owner}/{pkg} already installed.".format(owner=owner, pkg=pkg))
+            overwrite = input("Overwrite? (y/n) ")
+            if overwrite.lower() != 'y':
+                return
+        else:
+            pass
```
Comment (dimaryaz) [inline quilt/tools/command.py]: ?

File: quilt/tools/command.py [hunk 4]
```diff
@@ -410,10 +410,13 @@ def install(package, hash=None, version=None, tag=None):
     existing_pkg = store.get_package(owner, pkg)
 
     if existing_pkg is not None:
```
Comment (dimaryaz) [inline quilt/tools/command.py]: Just make it `if existing_pkg is not None and not force`. Or you could skip the `existing_pkg` lookup if `force=True`.

File: quilt/test/test_install.py (hunk 1 above)
Comment (akarve) [inline quilt/test/test_install.py]: how about a `pytest --integration` flag; off by default? and then we can decorate integration tests and skip without it.

File: quilt/test/test_install.py (hunk 2 above)
Comment (akarve) [inline quilt/test/test_install.py]: will probably end up splitting QuiltTestCase into two classes; one with file mocking; and subclass with scratch dir.

File: quilt/tools/command.py (hunk 3 above)
Comment (akarve) [inline quilt/tools/command.py]: i'm trying to rush and get article ready so many other things

Comment (akarve) [inline quilt/tools/command.py]: :hurtrealbad:

File: quilt/test/utils.py [hunk 5]
```diff
@@ -17,28 +17,36 @@
...
+
         self.session_patcher = patch('quilt.tools.command._get_session', requests.Session)
         self.session_patcher.start()
 
         self.requests_mock = responses.RequestsMock(assert_all_requests_are_fired=False)
         self.requests_mock.start()
 
     def tearDown(self):
-        self.requests_mock.stop()
+        super(QuiltTestCase, self).tearDown()
```
Comment (dimaryaz) [inline quilt/test/utils.py]: Just to be safe, let's do teardown in the reverse order - so call `super` at the end.

File: quilt/test/utils.py [hunk 6]
```diff
@@ -17,28 +17,36 @@
 import requests
 import responses
 
-
-class QuiltTestCase(unittest.TestCase):
+class QuiltTestCaseBasic(unittest.TestCase):
```
Comment (dimaryaz) [inline quilt/test/utils.py]: I'd prefer `BasicQuiltTestCase`, or something than ends in `TestCase`

**Pull request merged**